ompi/mpi/fortran/mpif-h/keyval_create_f.c

```diff
@@ -70,31 +70,34 @@ static const char FUNC_NAME[] = "MPI_keyval_create_f";
 
 void ompi_keyval_create_f(ompi_mpi1_fortran_copy_attr_function* copy_attr_fn,
                          ompi_mpi1_fortran_delete_attr_function* delete_attr_fn,
-                         MPI_Fint *keyval, MPI_Fint *extra_state,
+                         MPI_Fint *keyval_f, MPI_Fint *extra_state,
                          MPI_Fint *ierr)
 {
-    int ret, c_ierr;
+    int ret, c_ierr, keyval;
     ompi_attribute_fn_ptr_union_t copy_fn;
     ompi_attribute_fn_ptr_union_t del_fn;
 
     copy_fn.attr_mpi1_fortran_copy_fn = copy_attr_fn;
     del_fn.attr_mpi1_fortran_delete_fn = delete_attr_fn;
 
+    keyval = OMPI_FINT_2_INT(*keyval_f);
+
     /* Set the "F77_OLD" bit to denote that the callbacks should use
        the old MPI-1 INTEGER-parameter functions (as opposed to the
        new MPI-2 INTEGER(KIND=MPI_ADDRESS_KIND)-parameter
        functions). */
 
     ret = ompi_attr_create_keyval_fint(COMM_ATTR, copy_fn, del_fn,
-                                       keyval, *extra_state,
-                                       OMPI_KEYVAL_F77 | OMPI_KEYVAL_F77_MPI1,
+                                       &keyval, *extra_state,
+                                       OMPI_KEYVAL_F77,
```

Comment:
After reading the other commits, I *think* you removed it because you add it back in in the call to `ompi_attr_create_keyval_fint()` if `sizeof(int) == sizeof(INTEGER)`.  But:

1. These commits are out of order.
2. This particular flag refers to how attribute callbacks are invoked, not how the attributes are stored, right?